stdlib/public/core/DictionaryBuilder.swift

```diff
@@ -102,6 +102,13 @@ extension _NativeDictionary {
     ) -> Int
   ) {
     self.init(capacity: capacity)
+
+    // If the capacity is 0, then our storage is the empty singleton. Those are
+    // read only, so we shouldn't attempt to write to them.
+    if capacity == 0 {
+      return
```

Comment:
Marginal nit: We should still call the function argument exactly once, even if `capacity` happens to be zero.

```suggestion
      let c = initializer(
        UnsafeMutableBufferPointer(start: nil, count: 0), 
        UnsafeMutableBufferPointer(start: nil, count: 0))
      _precondition(c == 0)
      return
```